src/GitHub.InlineReviews/InlineCommentMarginProvider.cs

```diff
@@ -30,68 +30,59 @@ internal sealed class InlineCommentMarginProvider : IWpfTextViewMarginProvider
         readonly IViewTagAggregatorFactoryService tagAggregatorFactory;
         readonly IInlineCommentPeekService peekService;
         readonly IPullRequestSessionManager sessionManager;
+        readonly IPackageSettings packageSettings;
 
         [ImportingConstructor]
         public InlineCommentMarginProvider(
             IEditorFormatMapService editorFormatMapService,
             IViewTagAggregatorFactoryService tagAggregatorFactory,
             IInlineCommentPeekService peekService,
-            IPullRequestSessionManager sessionManager)
+            IPullRequestSessionManager sessionManager,
+            IPackageSettings packageSettings)
         {
             this.editorFormatMapService = editorFormatMapService;
             this.tagAggregatorFactory = tagAggregatorFactory;
             this.peekService = peekService;
             this.sessionManager = sessionManager;
+            this.packageSettings = packageSettings;
         }
 
         public IWpfTextViewMargin CreateMargin(IWpfTextViewHost wpfTextViewHost, IWpfTextViewMargin parent)
         {
             var textView = wpfTextViewHost.TextView;
             var editorFormatMap = editorFormatMapService.GetEditorFormatMap(textView);
             var glyphFactory = new InlineCommentGlyphFactory(peekService, textView, editorFormatMap);
-
-            Func<Grid> gridFactory = () => new GlyphMarginGrid();
-            return CreateMargin(glyphFactory, gridFactory, wpfTextViewHost, parent, editorFormatMap);
+            return CreateMargin(glyphFactory, wpfTextViewHost, parent, editorFormatMap);
         }
 
-        IWpfTextViewMargin CreateMargin<TGlyphTag>(IGlyphFactory<TGlyphTag> glyphFactory, Func<Grid> gridFactory,
+        IWpfTextViewMargin CreateMargin<TGlyphTag>(IGlyphFactory<TGlyphTag> glyphFactory,
             IWpfTextViewHost wpfTextViewHost, IWpfTextViewMargin parent, IEditorFormatMap editorFormatMap) where TGlyphTag : ITag
         {
             var tagAggregator = tagAggregatorFactory.CreateTagAggregator<TGlyphTag>(wpfTextViewHost.TextView);
-            var margin = new GlyphMargin<TGlyphTag>(wpfTextViewHost, glyphFactory, gridFactory, tagAggregator, editorFormatMap,
-                IsMarginVisible, MarginPropertiesName, MarginName, true, 17.0);
 
-            TrackCommentGlyphOnDiffView(wpfTextViewHost, margin.VisualElement);
-            return margin;
-        }
+            var isDiffView = IsDiffView(wpfTextViewHost);
+            var glyphMarginViewModel = new GlyphMarginViewModel(packageSettings, sessionManager, wpfTextViewHost, isDiffView);
```

Comment:
Should pass the `GlyphMarginViewModel` an instance of `InlineCommentTagger` for the view if one exists. The code creating the `InlineCommentTagger` should look at `IPackageSettings.EditorComments` when deciding to create a tagger (rather than doing this check in `GlyphMarginViewModel`). That way we can defer creating the TagAggregator until it's actually used.